bump minimum supported go version to 1.23

codeboten opened the pull request:
With the release of 1.24, we need to bump the tested versions.

mx-psi commented on .chloggen/codeboten_1.24.yaml:
```diff
@@ -0,0 +1,25 @@
+component: all
+
+# A brief description of the change.  Surround your text with quotes ("") if it needs to start with a backtick (`).
+note: Added support for go.14, bumped minimum version to 1.23
```
```suggestion
note: Added support for go 1.24, bumped minimum version to 1.23
```

codeboten replied:
```suggestion
note: Added support for go1.24, bumped minimum version to 1.23
```

evan-bradley commented on .chloggen/codeboten_1.24.yaml:
```diff
@@ -0,0 +1,25 @@
+component: all
+
+# A brief description of the change.  Surround your text with quotes ("") if it needs to start with a backtick (`).
+note: Added support for go.14, bumped minimum version to 1.23
```
```suggestion
note: Added support for go 1.24, bumped minimum version to 1.23
```

mx-psi reviewed (approved):
LGTM, we could try to do `1.23` instead of `1.23.0`, but I think we'll have to do `1.23.0` eventually, so I am not sure it is worth it

evan-bradley reviewed (approved):
There are also notes in the builder readme and in CONTRIBUTING.md that need to be updated.

codeboten commented:
>  There are also notes in the builder readme and in CONTRIBUTING.md that need to be updated.

@evan-bradley i updated the docs and removed the bits that only seemed relevant before 1.23 was the minimum, ptal

evan-bradley reviewed (approved):
Doc changes LGTM.
